### PR TITLE
apps/appgen: Add templates for a sample application of the task manager and the eventloop

### PR DESCRIPTION
**Pull request by tyha1115**
- Add templates for a sample application
1. Sample application using the task manager of TizenRT
2. Sample application using the eventloop of TizenRT

- Modify the appgen.sh in order to select the type of sample application.
- Revise the README.md according to the added templates and procedures.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [b6e6981f4a9d0cf3d7714513aeb76cb2e5aab282] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b6e6981f4a9d0cf3d7714513aeb76cb2e5aab282] - Code Rule Check OK.

File: apps/appgen/template_task_manager_sample_main.c_source [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#define SAMPLE_TASK_PERMISSION   TM_APP_PERMISSION_ALL
+#define SAMPLE_TASK_TIMEOUT      TM_RESPONSE_WAIT_INF
+
+/****************************************************************************
+ * Private Function
+ ****************************************************************************/
+
+static int sample_task(int argc, char *argv[])
+{
+	printf("Hello!, sample test is started by the task manager!!\n");
```
Comment (jeongchanKim) [inline apps/appgen/template_task_manager_sample_main.c_source]: sample test -> sample task

File: apps/appgen/template_task_manager_sample_main.c_source [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int ##ENTRY_FUNC##(int argc, char *argv[])
+#endif
+{
+	int handle;
+	int ret;
+
+	handle = task_manager_register_task("sample_task", SAMPLE_TASK_PRIORITY, SAMPLE_TASK_STACKSIZE, sample_task, SAMPLE_TASK_ARGV, SAMPLE_TASK_PERMISSION, SAMPLE_TASK_TIMEOUT);
```
Comment (jeongchanKim) [inline apps/appgen/template_task_manager_sample_main.c_source]: how about defining "sample_tas" as a #define?

**Comment by seinfra** [comment]
Target : [16b888c3905f03ba75d99ee2e91631d7d021c4c0] -

**Comment by seinfra** [comment]
Target : [16b888c3905f03ba75d99ee2e91631d7d021c4c0] -

**Comment by seinfra** [comment]
Target : [a47918b7f1dbfdb2fad473f47d6ed6b29ceac474] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a47918b7f1dbfdb2fad473f47d6ed6b29ceac474] - Code Rule Check (C++) OK.

File: apps/appgen/template_eventloop_sample_main.c_source [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+#endif
+{
+	int ret;
+
+	el_timer_t *timer;
+	el_event_t *event;
+	char *timer_cb_data = "Hello Timer";
+	char *event_cb_data = "Hello Event";
+
+	timer = eventloop_add_timer(EVENTLOOP_TIMER_INTERVAL, true, sample_timer_callback, timer_cb_data);
```
Comment (jeongchanKim) [inline apps/appgen/template_eventloop_sample_main.c_source]: is it okay not to check the return of timer?

File: apps/appgen/template_task_manager_sample_main.c_source [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+{
+	int ret;
+	tm_msg_t *broadcast_cb_data;
+
+	ret = task_manager_set_unicast_cb(sample_unicast_cb);
+	if (ret < 0) {
+		printf("Failed to set unicast callback! Return: %d\n", ret);
+	}
+
+	broadcast_cb_data = (tm_msg_t *)malloc(sizeof(tm_msg_t));
```
[the remark this answers is not in the document]
Comment (tyha1115) [reply apps/appgen/template_task_manager_sample_main.c_source]: Done.

File: apps/appgen/template_eventloop_sample_main.c_source (hunk 3 above)
Comment (tyha1115) [reply apps/appgen/template_eventloop_sample_main.c_source]: Done.

**Comment by seinfra** [comment]
Target : [6a123c59bdca5c2aabc9e4d677b60a9b7e46852f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6a123c59bdca5c2aabc9e4d677b60a9b7e46852f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [87b904c93ccf589f19b3f7f1ff0cb0748834cbf7] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6a10ff9994f9a5ed852996587a5d09476025d1e3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6a10ff9994f9a5ed852996587a5d09476025d1e3] - Code Rule Check (C++) OK.

**Pull request closed without merging**